frontend.java

```diff
@@ -0,0 +1,540 @@
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.ArrayList;
+
+
+public class frontend {
+	static int loggedIn = 0;
+	static int type = 0; // if type = 1 its retail, if type=2 its agent
+	static ArrayList<String> transactionFile = new ArrayList<String>();
+	public static void main(String[] args) {
+		login();
+	}
+	public static void login(){
+		//Runs on start up
+		
+				//Prompts User for log in
+				if(loggedIn == 0){
+					//Not logged in
+					while(loggedIn == 0){
```

Comment:
The `if(loggedIn == 0)` before the `while(loggedIn == 0)` is not needed.  If loggedIn=0, the condition of the while evaluates to true and we go into the loop.  If loggedIn=1, the condition of the while loop evaluates to false.

So there is an unnecessary if condition on line 22.